templates/power-off-vm.json

```diff
@@ -7,51 +7,84 @@
+                                "actions": {
+                                    "Send_an_email_(V2)": {
+                                        "inputs": {
+                                            "body": {
+                                                "Body": "<p><strong>FAILED <br>\n<br>\n</strong>Status: @{body('Get_virtual_machine')?['properties']?['provisioningState']}</p>",
+                                                "Subject": "Your automation task failed to stop virtual machine @{body('Get_virtual_machine')?['name']}",
+                                                "To": "@parameters('EmailID')"
+                                            },
+                                            "host": {
+                                                "connection": {
+                                                    "name": "@parameters('$connections')['office365']['connectionId']"
+                                                }
+                                            },
+                                            "method": "post",
+                                            "path": "/v2/Mail"
+                                        },
+                                        "runAfter": {},
+                                        "type": "ApiConnection"
+                                    }
+                                }
+                            },
+                            "expression": {
+                                "and": [
+                                    {
+                                        "equals": [
+                                            "@body('Get_virtual_machine')?['properties']?['provisioningState']",
+                                            "Succeeded"
+                                        ]
+                                    }
+                                ]
+                            },
+                            "runAfter": {},
+                            "type": "If"
                         }
                     },
                     "expression": {
-                        "or": [
+                        "and": [
                             {
                                 "equals": [
-                                    "@if(equals(string(body('Get_virtual_machine_state')?['properties']?['provisioningState']),'Succeeded'),'yes','no')",
-                                    "'yes'"
+                                    "@parameters('Notify')",
+                                    true
                                 ]
                             }
                         ]
                     },
                     "runAfter": {
-                        "Get_virtual_machine_state": [
-                            "Skipped",
-                            "Failed",
-                            "Succeeded"
+                        "Get_virtual_machine": [
+                            "Succeeded",
+                            "Failed"
```

Comment:
Didnt we decide to not have the 'Failed' state, since if it is failed you actually cannot get the outputs of this action

Reply:
When I actually did run it using the Logic app designer I did encounter a bug resulting from not having a run after for failed